classes/plugins/ThemePlugin.inc.php

```diff
@@ -275,13 +329,230 @@ public function &getScript($name) {
+			$context = Request::getContext();
+			$contextId = $context ? $context->getId() : 0;
+			$this->_optionValues = $pluginSettingsDAO->getPluginSettings($contextId, $this->getName());
+		}
+
+		if (isset($this->_optionValues[$name])) {
+			return $this->_optionValues[$name];
+		}
+
+		// Return a default if no value is set
+		$option = $this->getOptionConfig($name);
+		return $option && isset($option['default']) ? $option['default'] : null;
+	}
+
+	/**
+	 * Get an option's configuration settings
+	 *
+	 * This retrives option settings for any option attached to this theme or
+	 * any parent theme.
+	 *
+	 * @param $name The name of the option config to retrieve
+	 * @return false|array The config array for this option. Or false if no
+	 *  config is found.
+	 */
+	public function getOptionConfig($name) {
+
+		if (isset($this->options[$name])) {
+			return $this->options[$name];
+		}
+
+		return $this->parent ? $this->parent->getOptionConfig($name) : false;
+	}
+
+	/**
+	 * Get all options' configuration settings.
+	 *
+	 * This retrieves a single array containing options settings for this
+	 * theme and any parent themes.
+	 *
+	 * @return array
+	 */
+	public function getOptionsConfig() {
+
+		if (!$this->parent) {
+			return $this->options;
+		}
+
+		return array_merge(
+			$this->parent->getOptionsConfig(),
+			$this->options
+		);
+	}
+
+	/**
+	 * Get all option values
+	 *
+	 * This retrieves a single array containing option values for this theme
+	 * and any parent themes.
+	 *
+	 * @return array
+	 */
+	public function getOptionsValues() {
+
+		$pluginSettingsDAO = DAORegistry::getDAO('PluginSettingsDAO');
+
+		$context = Request::getContext();
+		$contextId = empty($context) ? 0 : $context->getId();
+		$values = $pluginSettingsDAO->getPluginSettings($contextId, $this->getName());
+
+		if (!$this->parent) {
+			return $values;
+		}
+
+		return array_merge(
+			$this->parent->getOptionsValues(),
+			$values
+		);
+	}
+
+	/**
+	 * Sanitize and save a theme option
+	 *
+	 * @param $name string A unique id for the option to save
+	 * @param $value mixed The new value to save
+	 * @param $contextId int Optional context id. Defaults to the current
+	 *  context
+	 */
+	public function saveOption($name, $value, $contextId = null) {
+
+		$option = !empty($this->options[$name]) ? $this->options[$name] : null;
+
+		if (is_null($option)) {
+			return $this->parent ? $this->parent->saveOption($name, $value) : false;
+		}
+
+		$type = '';
+		switch ($option['type']) {
+			case 'text' :
+            case 'select' :
+			case 'color' :
+				$type = 'text';
+				break;
+		}
+
+		if (is_null($contextId)) {
+			$context = Request::getContext();
+			$contextId = $context->getId();
+		}
+
+		$this->updatesetting($contextId, $name, $value, $type);
+
+		// Clear the template cache so that new settings can take effect
+		$templateMgr = TemplateManager::getManager($this->getRequest());
+		$templateMgr->clearTemplateCache();
```

Comment:
I think we do if we load separate asset requests based on the options (eg - different requests to Google fonts).

Either way, future theme options could be used to adjust actual markup, and could be queried from within template files. So I'd rather start off on the right foot.